sandbox-tools/merge-repos/src/mergeStagingToMain.ts

```diff
@@ -0,0 +1,1045 @@
+import { getUser } from "./git/userDetails";
+import { checkPrExists } from "./github/checkPrExists";
+import { createPullRequest, gitHubCreateForkRepo } from "./github/github";
+import { createPackageKarmaConfig } from "./tests/karma";
+import { createPackageRollupConfig } from "./rollup/rollup";
+import { rushUpdateShrinkwrap } from "./rush/rush";
+import { abort, fail, terminate } from "./support/abort";
+import { addCleanupCallback, doCleanup } from "./support/clean";
+import { isIgnoreFolder } from "./support/isIgnoreFolder";
+import { moveFolder } from "./support/moveTo";
+import { parseArgs, ParsedOptions, SwitchBase } from "./support/parseArgs";
+import { IMergeDetail, IMergePackageDetail, IPackageJson, IRepoDetails, IRushJson } from "./support/types";
+import { dumpObj, findCurrentRepoRoot, formatIndentLines, log, logError, logWarn, removeTrailingComma, transformPackages } from "./support/utils";
+import { createPackageWebpackTestConfig } from "./tests/webpack";
+
+interface IStagingRepoDetails {
+    git: SimpleGit,
+    path: string,
+    branch: string,
+    commitDetails: ICommitDetails
+}
+
+interface IPackageJsonDetail {
+    isNew?: boolean;
+
+    /**
+     * The root path for the project
+     */
+    path: string,
+
+    /**
+     * The relative path for the project
+     */
+    rPath: string,
+
+     /**
+     * The path to the package.json
+     */
+    pkgPath: string,
+
+    /**
+     * Holds the loaded package.json
+     */
+    pkg: IPackageJson,
+
+    /**
+     * The original package.json text
+     */
+    pkgText: string
+}
+
+interface IPackages {
+    src: { [key: string]: IPackageJsonDetail },
+    dest: { [key: string]: IPackageJsonDetail }
+}
+
+/**
+ * The command line options for this script
+ */
+ interface MergeStagingToMainOptions extends SwitchBase {
+    cloneTo: string;
+    originRepo: string;
+    stagingBranch: string;
+    destBranch: string;
+
+    stagingStartPoint?: string;                          // Used for local testing to validate periodic execution
+
+    /**
+     * Run the script but don't create the final PR and move the destination folder up one level
+     * ie. prefix "../" to the cloneTo location
+     */
+    test: boolean;
+
+    /**
+     * Don't create the PR
+     */
+    noPr: boolean;
+}
+
+// The current git repo root
+const _gitRoot = findCurrentRepoRoot();
+
+// The path that will be used for the merge git root (clone / working folder)
+let _mergeGitRoot: string;
+
+// Holds the current branch, so it can be restore if necessary on normal exit
+let _currentBranch: string;
+
+/**
+ * The default command line arguments
+ */
+let _theArgs: ParsedOptions<MergeStagingToMainOptions> = {
+    failed: false,
+    values: [],
+    switches: {
+        cloneTo: MERGE_CLONE_LOCATION,
+        originRepo: MERGE_ORIGIN_REPO,
+        stagingBranch: MERGE_ORIGIN_STAGING_BRANCH,
+        destBranch: MERGE_ORIGIN_MERGE_MAIN_BRANCH,
+        test: true,
```

Comment:
I have this command line flag currently set to true for local testing (without having to pass it every time), this will be false (as there is now way to specify false from the command line with the way the parser is constructed) in the `support/parseArgs.ts`